src/platform/plugins/shared/discover/public/application/main/state_management/redux/actions/reset_discover_session.test.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { cloneDeep } from 'lodash';
+import { BehaviorSubject } from 'rxjs';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { DiscoverStateContainer } from '../../discover_state';
+import { createDiscoverServicesMock } from '../../../../../__mocks__/services';
+import { getDiscoverStateMock } from '../../../../../__mocks__/discover_state.mock';
+import type { DiscoverSessionTab } from '@kbn/saved-search-plugin/common';
+import type { SavedSearch } from '@kbn/saved-search-plugin/public';
+import { internalStateActions } from '..';
+import { internalStateSlice } from '../internal_state';
+import { fromTabStateToSavedObjectTab } from '../tab_mapping_utils';
+import { getTabStateMock } from '../__mocks__/internal_state.mocks';
+import { getTabRuntimeStateMock } from '../__mocks__/runtime_state.mocks';
+import * as tabsActions from './tabs';
+import type { TabState } from '../types';
+import { dataViewMock, dataViewMockWithTimeField } from '@kbn/discover-utils/src/__mocks__';
+import { createSearchSourceMock } from '@kbn/data-plugin/public/mocks';
+
+interface SetupOptions {
+  additionalPersistedTabs?: DiscoverSessionTab[];
+  savedSearch?: SavedSearch | false;
+}
+
+const servicesMock = createDiscoverServicesMock();
+
+const createPersistedTab = () =>
+  fromTabStateToSavedObjectTab({
+    tab: getTabStateMock({
+      id: 'secondary-tab',
+      label: 'Secondary tab',
+      initialInternalState: {
+        serializedSearchSource: { index: dataViewMock.id },
+      },
+    }),
+    timeRestore: false,
+    services: servicesMock,
+  });
+
+const ensureRuntimeState = (state: DiscoverStateContainer, tabId: string) => {
+  const existingTab = state.runtimeStateManager.tabs.byId[tabId];
```

Comment:
Nit: you could use `selectTabRuntimeState` for this, although not especially important in a test file.